File: walk.py
#!/usr/bin/env python

import os
import sys

# Set the directory you want to start from
rootDir = sys.argv[1]
gitpath = f'{rootDir}/.git'


# https://docs.python.org/3.3/library/os.html?highlight=os.walk#os.walk
for dirpath, dirnames, filenames in os.walk(rootDir):
    if dirpath == gitpath:
        dirnames[:] = []
        continue
    dirnames.sort()
    filenames.sort()
    for fname in filenames:
        print(f'{dirpath}/{fname}')

File: walk.pl
#!/usr/bin/env perl

use warnings;
use strict;
use 5.010;

use File::Next;

my $start = shift or die 'Specify a directory to walk';

my $iter = File::Next::files(
    {
        descend_filter => sub { $_ ne '.git' },
        sort_files     => 1,
    },
    $start
);

my %seen;
while ( my $file = $iter->() ) {
    #    ++$seen{$file};
    say $file;
}
